test/index.js

```diff
@@ -1338,3 +1338,26 @@ describe('parse', function() {
   });
 
 });
+
+describe('regex', function() {
+
+  describe('match', function() {
+    var str = 'abcdefg';
+    var search = /abcd/;
+    var result = ['abcd'];
+
+    it('returns a Maybe', function() {
+      eq(S.fromMaybe({}, S.match(search, str)).toString(), result.toString());
```

Comment:
We could remove the use of `S.fromMaybe`. I'd also prefer literals in place of `search`, `str`, and `result.toString()`, so the assertion requires no context. I would write:

```javascript
eq(S.match(/abcd/, 'abcdefg').toString(), 'Just(["abcd"])');
```